src/json_schema/validators/of.rs

```diff
@@ -12,17 +12,56 @@ pub struct AllOf {
 impl super::Validator for AllOf {
     fn validate(&self, val: &Value, path: &str, scope: &scope::Scope) -> super::ValidationState {
         let mut state = super::ValidationState::new();
+        let mut val = Cow::Borrowed(val);
+
+        // first get all relevant schemas
+        let schemas = self
+            .schemes
+            .iter()
+            .map(|url| (url, scope.resolve(url)))
+            .filter_map(|(url, opt)| {
+                if opt.is_none() {
+                    state.missing.push(url.clone())
+                }
+                opt
+            })
+            .collect::<Vec<_>>();
+
+        // first pass to populate all defaults (if enabled)
+        for schema in schemas.iter() {
+            let mut result = schema.validate_in(&val, path);
+            if result.is_valid() && result.replacement.is_some() {
+                *val.to_mut() = result.replacement.take().unwrap();
```

Comment:
Shouldn't all possible replacements be merged into `val` instead of it being overwritten?

Reply:
Yes, indeed that is the case: notice how the updated `&val` will be passed to the next schema two lines up.